contracts/src/v0.8/dev/ArbitrumValidator.sol

```diff
@@ -240,52 +324,111 @@ contract ArbitrumValidator is TypeAndVersionInterface, AggregatorValidatorInterf
       return true;
     }
 
-    int isServiceOffline = 1;
-    // NOTICE: if gasCostL2 is zero the payment is processed on L2 so the L2 address needs to be funded, as it will
-    // paying the fee. We also ignore the returned msg number, that can be queried via the InboxMessageDelivered event.
-    s_inbox.createRetryableTicket{value: s_gasConfig.gasCostL2}(
-      s_l2FlagsAddress,
+    // Excess gas on L2 will be sent to the L2 xDomain alias address of this contract
+    address refundAddr = L2_ALIAS;
+    // Encode the Forwarder call
+    bytes4 selector = ForwarderInterface.forward.selector;
+    address target = L2_FLAGS;
+    // Choose and encode the underlying Flags call
+    bytes memory data = currentAnswer == ANSWER_SEQ_OFFLINE ? CALL_RAISE_FLAG : CALL_LOWER_FLAG;
+    bytes memory message = abi.encodeWithSelector(selector, target, data);
+    // Make the xDomain call
+    // NOTICE: We approximate the max submission cost of sending a retryable tx with specific calldata length.
+    uint256 maxSubmissionCost = _approximateMaxSubmissionCost(message.length);
+    uint256 maxGas = s_gasConfig.maxGas;
+    uint256 gasPriceBid = s_gasConfig.gasPriceBid;
+    uint256 l1PaymentValue = s_paymentStrategy == PaymentStrategy.L1 ? _maxRetryableTicketCost(maxSubmissionCost, maxGas, gasPriceBid) : 0;
+    // NOTICE: In the case of PaymentStrategy.L2 the L2 xDomain alias address needs to be funded, as it will be paying the fee.
+    // We also ignore the returned msg number, that can be queried via the `InboxMessageDelivered` event.
+    IInbox(CROSS_DOMAIN_MESSENGER).createRetryableTicketNoRefundAliasRewrite{value: l1PaymentValue}(
+      L2_CROSS_DOMAIN_FORWARDER, // target
       0, // L2 call value
-      // NOTICE: maxSubmissionCost info will possibly become available on L1 after the London fork. At that time this
-      // contract could start querying/calculating it directly so we wouldn't need to configure it statically. On L2 this
-      // info is available via `ArbRetryableTx.getSubmissionPrice`.
-      s_gasConfig.maxSubmissionCost, // Max submission cost of sending data length
-      s_gasConfig.refundableAddress, // excessFeeRefundAddress
-      s_gasConfig.refundableAddress, // callValueRefundAddress
-      s_gasConfig.gasLimitL2,
-      s_gasConfig.maxGasPrice,
-      currentAnswer == isServiceOffline ? CALL_RAISE_FLAG : CALL_LOWER_FLAG
+      maxSubmissionCost,
+      refundAddr, // excessFeeRefundAddress
+      refundAddr, // callValueRefundAddress
+      maxGas,
+      gasPriceBid,
+      message
     );
+    // return success
```

Comment:
Would an event be useful here? As on `withdrawFundsFromL2` with `emit L2WithdrawalRequested(id, amount, refundAddr);`

Reply:
Crossed my mind, but potentially an overkill to add another event between the aggregator's `Transmit` one and the bridge `InboxMessageDelivered` (`SentMessage` on Optimism). Tracking `Transmit` events should give us all the required info.

Do you have a use case in mind?

Reply:
The only case I was thinking it was tracking the message, but `InboxMessageDelivered` does it already then